crates/services/relayer/tests/integration.rs

```diff
@@ -124,6 +127,40 @@ async fn can_get_messages() {
     }
 }
 
+#[tokio::test(start_paused = true)]
+async fn can_get_transactions() {
+    let mock_db = MockDb::default();
+    let eth_node = MockMiddleware::default();
+
+    let config = Config::default();
+    let contract_address = config.eth_v2_listening_contracts[0];
+    let transaction = |max_gas: u64, block_number: u64| {
+        let transaction = TransactionFilter {
+            max_gas,
+            ..Default::default()
+        };
+        let mut log = transaction.into_log();
+        log.address = contract_address;
+        log.block_number = Some(block_number.into());
+        log
+    };
+
+    let logs = vec![transaction(2, 1), transaction(3, 2)];
+    let expected_transactions: Vec<_> = logs.iter().map(|l| l.to_tx()).collect();
+    eth_node.update_data(|data| data.logs_batch = vec![logs.clone()]);
+    // Setup the eth node with a block high enough that there
+    // will be some finalized blocks.
+    eth_node.update_data(|data| data.best_block.number = Some(100.into()));
+    let relayer = new_service_test(eth_node, mock_db.clone(), config);
+    relayer.start_and_await().await.unwrap();
+
+    relayer.shared.await_synced().await.unwrap();
+
+    for tx in expected_transactions {
+        assert_eq!(mock_db.get_transaction(&tx.relayed_id()).unwrap(), tx);
```

Comment:
I think we should make it clear that `start_and_await` is the SUT here:

Given an eth node with these logs,
when we run `start_and_await`,
then those logs are picked up and stored in the db as txs